### PR TITLE
codeql changes

### PR DESCRIPTION
**Pull request by gcampbell-msft**
Fix CodeQL bugs.

### CODE REVIEW

File: Source/LexerParser/cmListFileLexer.c [hunk 1]
```diff
@@ -2722,10 +2722,12 @@ static cmListFileLexer_BOM cmListFileLexer_ReadBOM(FILE* f)
       }
     } else if (b[0] == 0xFF && b[1] == 0xFE) {
       fpos_t p;
-      fgetpos(f, &p);
       if (fread(b, 1, 2, f) == 2 && b[0] == 0 && b[1] == 0) {
         return cmListFileLexer_BOM_UTF32LE;
       }
+	  if (fgetpos(f, &p) != 0) {
```
Comment (benmcmorran) [inline Source/LexerParser/cmListFileLexer.c]: Does it matter that this call is now after `fread()`, which could affect the position?

Comment (gcampbell-msft) [reply Source/LexerParser/cmListFileLexer.c]: Correct me if I'm wrong, but it shouldn't based on my understanding of the code. 

`p` isn't used anywhere except for the `fgetpos` and `fsetpos` calls, and based on my reading of the cpp docs regarding `fread`, it doesn't take in a position parameter, it simply reads the buffer. 

Does that seem right?

Comment (gcampbell-msft) [reply Source/LexerParser/cmListFileLexer.c]: Ohh, but I see what you're saying, `fread` could change it. Yes I think it's possible, I'll slightly refactor.

Comment (gcampbell-msft) [reply Source/LexerParser/cmListFileLexer.c]: Refactored.

**Review by benmcmorran — APPROVED** [review]

**Pull request merged**